extensions/extensions-docker/aissemble-configuration-store/src/main/resources/scripts/genWebhook.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+###
+# #%L
+# aiSSEMBLE::Extensions::Docker::Configuration Store
+# %%
+# Copyright (C) 2021 Booz Allen
+# %%
+# This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+# #L%
+###
+
+set -e
+SCRIPT=`basename ${BASH_SOURCE[0]}`
+
+function usage {
+  cat<< EOF
+  Usage: $SCRIPT
+  Options:
+  -h | --help                    Display help information.
+  -c | --ca-bundle <ca-bundle>   PEM encoded CA bundle which will be used to validate the webhook's server certificate.
+  -n | --namespace <namespace>   The namespace where the the config store is installed.			
+  -s | --service <service>       The name of the config store webhook service.
+  -w | --webhook-name <name>     The name of the webhook for the configuration store.
+EOF
+}
+
+function parse_arguments {
+  while [[ $# -gt 0 ]]
+  do
+    case "$1" in
+      -c|--ca-bundle)
+      if [[ -n "$2" ]]; then
+        CA_BUNDLE="$2"
+      else
+        echo "-c or --ca-bundle requires a value."
+        exit 1
+      fi
+      shift 2
+      continue
+      ;;
+      -n|--namespace)
+      if [[ -n "$2" ]]; then
+        NAMESPACE="$2"
+      else
+        echo "-n or --namespace requires a value."
+        exit 1
+      fi
+      shift 2
+      continue
+      ;;
+      -s|--service)
+      if [[ -n "$2" ]]; then
+        SERVICE="$2"
+      else
+        echo "-s or --service requires a value."
+        exit 1
+      fi
+      shift 2
+      continue
+      ;;
+      -w|--webhook-name)
+      if [[ -n "$2" ]]; then
+        WEBHOOK_NAME="$2"
+      else
+        echo "-w or --webhook-name requires a value."
+        exit 1
+      fi
+      shift 2
+      continue
+      ;;
+      -h|--help)
+      usage
+      exit 0
+      ;;
+      --)              # End of all options.
+        shift
+        break
+      ;;
+      '')              # End of all options.
+        break
+      ;;
+      *)
+        echo "Unrecognized option: $1"
+        exit 1
+      ;;
+    esac
+  done
+}
+
+parse_arguments "$@"
+
+echo "Creating a MutatingWebhookConfiguration"
+
+TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+
+TMP_DIR="/tmp/config-store-webhook/"
+mkdir -p ${TMP_DIR}
+
+# Base64 encode and then remove the any line breaks to avoid issues in the curl command
+ca_bundle=$(echo "$CA_BUNDLE"| base64 -w 0| tr -d '\n')
```

Comment:
Q: Wouldn't the CA bundle already be b64 encoded?

Reply:
Interestingly when it gets passed in from the env it get un-encoded. So we have to re-encode again here.